packages/repository/test/unit/query/query-builder.unit.ts

```diff
@@ -139,10 +147,44 @@ describe('WhereBuilder', () => {
   });
 
   it('constrains an existing where object with another where filter', () => {
-    const builder = new WhereBuilder({x: 'x'});
+    const builder = new WhereBuilder<AnyObject>({x: 'x'});
     const where = builder.impose({x: 'y', z: 'z'}).build();
     expect(where).to.be.deepEqual({and: [{x: 'x'}, {x: 'y', z: 'z'}]});
   });
+
+  it('compiles valid where clause', () => {
+    interface Post {
+      id: number;
+      title: string;
+      author: string;
+      likes: number;
+      gte: number;
+      regexp: unknown;
+    }
+
+    type Key = KeyOf<Post>;
+
+    const k1: Key = 'id';
+    // const k2: Key = 'regexp'; // fail to compile, regexp is a keyword
+    // const k3: Key = 'gte'; // fail to compile, gte is a keyword
```

Comment:
I guess we don't really have a way to "catch" compiler errors, but if these were to be uncommented, we can see them right?

Reply:
Yes